### PR TITLE
[LW-10190] Revamp multi wallet create and restore flows

### PR DESCRIPTION
**Pull request by shawnbusuttil**
# Checklist

- [ ] JIRA - \<LW-10249>
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

- Revamping mnemonic stages of multi wallet.
- Deprecating unused screens.

## Testing

Full acceptance criteria on LW-10249 ticket. Go through create and restore multi wallet flows.

## Screenshots

Attach screenshots here if implementation involves some UI changes

### CODE REVIEW

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx [hunk 1]
```diff
@@ -64,33 +71,43 @@ export const NewRecoveryPhrase = (): JSX.Element => {
...
         passphraseInfoLink={`${process.env.FAQ_URL}?question=what-happens-if-i-lose-my-recovery-phrase`}
+        onWatchVideoClick={() =>
+          analytics.sendEventToPostHog(postHogOnboardingActions.create.RECOVERY_PHRASE_INTRO_WATCH_VIDEO_CLICK)
+        }
+        onCopyToClipboard={() =>
+          analytics.sendEventToPostHog(postHogOnboardingActions.create.RECOVERY_PHRASE_COPY_TO_CLIPBOARD_CLICK)
+        }
+        onPasteFromClipboard={() =>
+          analytics.sendEventToPostHog(postHogOnboardingActions.create.RECOVERY_PHRASE_PASTE_FROM_CLIPBOARD_CLICK)
+        }
```
Comment (greatertomi) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx:109]: We should use a different analytics event from the onboarding. Maybe we can create stubs of analytics constants with no events. There is a ticket to address analytics later and can just fill in these.

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/SelectRecoveryPhraseLength.tsx [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
-export const SelectRecoveryPhraseLength = (): JSX.Element => {
-  const { t } = useTranslation();
-  const history = useHistory();
-  const { setLength } = useRestoreWallet();
-  const analytics = useAnalyticsContext();
-
-  const walletSetupRecoveryPhraseLengthStepTranslations = {
-    title: t('core.walletSetupRecoveryPhraseLengthStep.title'),
-    description: t('core.walletSetupRecoveryPhraseLengthStep.description'),
-    wordPassphrase: t('core.walletSetupRecoveryPhraseLengthStep.wordPassphrase')
```
Comment (greatertomi) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/SelectRecoveryPhraseLength.tsx:23]: We should delete these from the translations file

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx [hunk 3]
```diff
@@ -64,33 +71,43 @@ export const NewRecoveryPhrase = (): JSX.Element => {
...
-              break;
-            case PASSPHRASE_STEP_3:
-              stage === 'input'
-                ? analytics.sendEventToPostHog(PostHogAction.MultiwalletCreateEnterPassphrase17NextClick)
-                : analytics.sendEventToPostHog(PostHogAction.MultiwalletCreateWritePassphrase17NextClick);
-          }
+        onStepNext={(mnemonicStage: MnemonicStage) => {
+          mnemonicStage === 'writedown'
+            ? analytics.sendEventToPostHog(postHogOnboardingActions.create.SAVE_RECOVERY_PHRASE_NEXT_CLICK)
+            : analytics.sendEventToPostHog(postHogOnboardingActions.create.ENTER_RECOVERY_PHRASE_NEXT_CLICK);
```
Comment (greatertomi) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx:96]: Same analytics comment as here.

**Review by greatertomi — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx (hunk 1 above)
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx:109]: Yes arent these in scope of another ticket?

**Review by greatertomi — APPROVED** [review]

**Review by oldGreg5 — APPROVED** [review]

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx [hunk 4]
```diff
@@ -10,13 +10,10 @@ import { useWalletManager } from '@hooks/useWalletManager';
 import { useAnalyticsContext } from '@providers/AnalyticsProvider';
 import { PostHogAction } from '@lace/common';
 import { getWalletAccountsQtyString } from '@src/utils/get-wallet-count-string';
+import { postHogOnboardingActions } from '@providers/AnalyticsProvider/analyticsTracker';
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx:13]: I am not 100% sure about that, but I don't think we should use the onboarding events for the multiwallet flows. There are dedicated multiwallet events, and they should be rather adjusted to be aligned with the onboarding events. Could you please double check that? 🙏

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx [hunk 5]
```diff
@@ -64,33 +71,43 @@ export const NewRecoveryPhrase = (): JSX.Element => {
 
   return (
     <>
-      <WalletSetupMnemonicStep
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx:67]: Please remove the `WalletSetupMnemonicStep` as it is not used anywhere now.

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/RestoreRecoveryPhrase.tsx [hunk 6]
```diff
@@ -40,68 +44,64 @@ export const RestoreRecoveryPhrase = (): JSX.Element => {
...
+  }, [data, clearSecrets, createWallet, history, t, analytics, walletRepository]);
+
+  const handleMnemonicVerification = (event: Readonly<React.MouseEvent<HTMLButtonElement>>) => {
+    event.preventDefault();
+    analytics.sendEventToPostHog(postHogOnboardingActions.restore?.ENTER_RECOVERY_PHRASE_NEXT_CLICK);
+    onSubmitForm();
+  };
 
   return (
-    <WalletSetupMnemonicVerificationStep
```
Comment (szymonmaslowski) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/RestoreRecoveryPhrase.tsx:82]: Please remove the old component as it is no longer used.

File: packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicStepRevamp.tsx [hunk 7]
```diff
@@ -109,9 +108,9 @@ export const WalletSetupMnemonicStepRevamp = ({
         >
           {translations.writePassphraseSubtitle2}
         </span>
-        <Dialog.Root open={videoModalOpen} setOpen={setVideoModalOpen}>
+        <Modal open={videoModalOpen} centered footer={undefined} closable={false} destroyOnClose maskClosable={false}>
```
Comment (szymonmaslowski) [inline packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicStepRevamp.tsx:111]: `Dialog.Root` has a `zIndex` prop. Modal from antd applies a `z-index` of 1000 so we need to set it to a higher number.
```suggestion
        <Dialog.Root open={videoModalOpen} setOpen={setVideoModalOpen} zIndex={1001}>
```
I don;t like that it is being passed in the markup but this is how it is implemented now :/. Despite that I would still prefer to use lace ui as the goal is to introduce it gradually and eliminate antd at some point.

We could also change the root modal of the multiwallet to lace-ui's one and that would also eliminate a problem without setting the zIndex, but I don't request it here 😉

**Review by szymonmaslowski — CHANGES_REQUESTED** [review]
Great job!

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx (hunk 4 above)
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/steps/NewRecoveryPhrase.tsx:13]: They will be addressed in the follow up tasks we have regarding posthog.

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/components/index.ts [hunk 8]
```diff
@@ -1,2 +0,0 @@
-export { Home } from './Home';
```
[the remark this answers is not in the document]
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/components/index.ts:1]: There aren't any export issues. It seems like the dependents are still managing to import it.

File: packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicStepRevamp.tsx (hunk 7 above)
Comment (shawnbusuttil) [reply packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicStepRevamp.tsx]: Resolved

File: packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicStepRevamp.tsx [hunk 9]
```diff
@@ -109,9 +109,12 @@ export const WalletSetupMnemonicStepRevamp = ({
         >
           {translations.writePassphraseSubtitle2}
         </span>
-        <Dialog.Root open={videoModalOpen} setOpen={setVideoModalOpen}>
+        <Dialog.Root open={videoModalOpen} setOpen={setVideoModalOpen} zIndex={1001}>
           {renderVideoPopupContent({ onClose: () => setVideoModalOpen(false) })}
         </Dialog.Root>
+        {/* <Modal open={videoModalOpen} centered footer={undefined} closable={false} destroyOnClose maskClosable={false}>
```
Comment (szymonmaslowski) [inline packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicStepRevamp.tsx:115]: Please remove this commented code.

**Review by szymonmaslowski — APPROVED** [review]
GJ 🚀 Approving, but please apply the last comment.

**Pull request merged**